pkg/hash/hash.go

```diff
@@ -12,14 +12,20 @@ func TemplateTierHashLabelKey(tierName string) string {
 	return toolchainv1alpha1.LabelKeyPrefix + tierName + "-tier-hash"
 }
 
-// ComputeHashForNSTemplateTier computes the hash of the `.spec.namespaces[].templateRef` + `.spec.clusteResource.TemplateRef`
+// ComputeHashForNSTemplateTier computes the hash of the value of `status.revisions[]`
+// currently it passes the `.spec.namespaces[].templateRef` in key field and `.spec.clusteResource.TemplateRef` value in Key field.
+// as we want to make sure that we just use the values of `.spec.namespaces[].templateRef`+ `.spec.clusteResource.TemplateRef`
+// and we do not use the extra values available in `status.revisions[]` since there is no logic yet to delete the extra key-value pairs
+// if the extra values are used while calculating hash, it won't be equal to the hash of `NSTemplateSetSpec`
+// TODO : once there is logic to have the `status.revisions[]` cleaned up, update this function to just loop over `status.revisions[]`
+// to calculate hash
 func ComputeHashForNSTemplateTier(tier *toolchainv1alpha1.NSTemplateTier) (string, error) {
 	refs := []string{}
 	for _, ns := range tier.Spec.Namespaces {
-		refs = append(refs, ns.TemplateRef)
+		refs = append(refs, tier.Status.Revisions[ns.TemplateRef])
 	}
 	if tier.Spec.ClusterResources != nil {
-		refs = append(refs, tier.Spec.ClusterResources.TemplateRef)
+		refs = append(refs, tier.Status.Revisions[tier.Spec.ClusterResources.TemplateRef])
```

Comment:
As it is mentioned in the Jira and also in the design document, we should use the whole revisions map to collect the refs and compute the hash, not only specific references.
It was also mentioned in the other threads that the current logic has a bug that it doesn't include the refs of the role templates - this should be fixed by collecting all revisions from the map

Reply:
Currently we have extra values  and we do not have the logic of delete/clean up of revisions field, thats the reason i did this https://github.com/codeready-toolchain/toolchain-common/pull/454/files#diff-1da09bec2a4b26b257932b06c7ea839cc44492e089bd8276e4ae348bc6772b78R20

if i dont do it and if there are extra values, it will mess with the hash , thats my understanding .
Thats the reason i did the specific check for now and then probably change it once we have the clean-up logic for revisions.
WDYT?